Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -74,13 +78,68 @@ Ref<ViewTimeline> ViewTimeline::createFromCSSValue(Style::BuilderState& builderS
     return adoptRef(*new ViewTimeline(nullAtom(), axis, { startInset, endInset }));
 }
 
-// FIXME: compute offset values from options.
+static std::optional<Length> lengthForInset(std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>> inset)
+{
+    // TODO: Need to test this
+    if (auto* numericInset = std::get_if<RefPtr<CSSNumericValue>>(&inset)) {
+        if (RefPtr insetValue = dynamicDowncast<CSSUnitValue>(*numericInset)) {
+            auto length = insetValue->convertTo(CSSUnitType::CSS_PX);
+            if (!length)
+                return { };
+            return Length(length->value(), LengthType::Fixed);
+        }
+    }
+    ASSERT(std::holds_alternative<RefPtr<CSSKeywordValue>>(inset));
+    return { };
+}
+
+static Length lengthForInsetCSSValue(RefPtr<CSSPrimitiveValue> value)
+{
+    // TODO: Figure out how to make this work when provided calc value
+    if (!value || value->isCalculated())
+        return { };
+    if (value->valueID() == CSSValueAuto)
+        return Length();
```

Comment:
Would be better to stick to `Length()` or `{ }` for consistency.

Reply:
Fixed.